src/components/language-switch/language-switch.tsx

```diff
@@ -1,44 +1,91 @@
 import { useI18next } from 'gatsby-plugin-react-i18next';
 import React from 'react';
 import styled from 'styled-components';
-import { Link } from '../links/links';
+import { Chevron } from '../icons/chevron';
 
-const StyledLanguageLink = styled(Link)<{ selected?: boolean }>`
-  margin: 1px 0 1px 12px;
+interface LanguageSwitchProps {
+  translation: any;
+  className?: string;
+  $lightTheme?: boolean;
+  fromNavigation?: boolean;
+}
+
+const StyledNav = styled.nav<{
+  $lightTheme?: boolean;
+  fromNavigation?: boolean;
+}>`
+  transition: color 0.2s;
+  color: ${(props) =>
+    props.fromNavigation || props.$lightTheme
+      ? props.theme.palette.text.default
+      : props.theme.palette.text.header};
+  &:hover {
+    color: ${(props) =>
+      props.fromNavigation
+        ? props.theme.palette.text.header
+        : !props.$lightTheme &&
+          !props.fromNavigation &&
+          props.theme.palette.text.headerHover};
+  }
+`;
```

Comment:
I know that's styled component but it's a mess to read and understand 😬
Do we have better patterns to handle a "complex" data structure like this? Like "calculated" theme values? You could use local methods in the tsx file for that purpose. This will make it much, much more readable.

This is a suggestion which you can implement within this PR or afterwards. But I would really like to see this addressed. Even if we did it like that in the past, I think we should find a better pattern. I hope you agree.

Maybe @fischeversenker or @feedm3 have some opinion in that too?

```
/**
Idea: calculate the theme colors outside of the style definition. Do not use ternary operator.
If possible comment if it's not clear what's happening too.
**/
const getTextColor(props) => {
   if(props.fromNavigation || props.$lightTheme) {
      return props.theme.palette.text.default
   }else {
      return props.theme.palette.text.header
   }
}


const getTextHoverColor(props) => {
   // ....
}


const StyledNav = styled.nav
// ....
color: ${(props) => getTextColor(props)
&:hover {
    color: ${(props) => getTextHoverColor(props)
```

Reply:
I agree with you, I didn't like that part during development either, but I couldn't think of a better solution either. A local method is definitely better 👍

Reply:
Agree. Make sure to use types that it's not guessing which props are available ;-)